### ISSUE TITLE
Convert `targethost` to allow for clusters or targethosts

### ISSUE THREAD

**Issue by jjasghar** [chef/kitchen-vcenter#9]
Changing the `targethost` to the high level "cluster" causes the following error:

```
~/repo/vmworld2017-demo/vim on master [d]± be kitchen verify ubuntu
-----> Starting Kitchen (v1.16.0)
-----> Creating <default-ubuntu-1604>...
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Failed to complete #create action: [Unable to find target host: Cluster] on default-ubuntu-1604
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration

~/repo/vmworld2017-demo/vim on master [d]±
```

Though if i point it to a specific host _in_ the cluster it works fine, we should have some logic to either pick a machine in the cluster, do a specific cluster, or maybe even let DRS do it for you.

**Comment by tecracer-theinen** [comment]
Just as a note:

Provisioning a VM into a specific cluster is equivalent of provisioning it into the default resource pool "Resources" there. Thus, the code will need to lookup the cluster via API, get the ID of the default pool and use that for going on with provisioning.

**Comment by siemensgamesa-theinen** [comment]
Can be closed: Published with kitchen-vcenter v1.4.1 / PR #42
